app/controllers/membership_applications_controller.rb

```diff
@@ -126,20 +138,32 @@ def authorize_membership_application
   end
 
 
-  def new_upload_file(upload_file_param)
-    if upload_file_param['actual_files']
-      upload_file_param['actual_files'].each do |upload_file|
+  def new_file_uploaded(params)
+
+    successful = true
+
+    if params.fetch('uploaded_file', false) && params['uploaded_file'].fetch('actual_files', false)
+
+      params['uploaded_file']['actual_files'].each do |uploaded_file|
```

Comment:
I guess I'm so used to that pattern that it's plain to me.  IOW, I'm blind to the problem.

We don't know if the 'uploaded_file' key will exists in `params`, and if it does, we don't know if `actual_files` will exist with that sub-hash (at `params['uploaded_file']`).  Only if both of those conditions are met can we loop thru each `uploaded_file`.
Hence I check using `.fetch` (and return `false` if a key doesn't exist).

Should I just put that into a comment in the code?  Or is there something else /additional that should be clarified?

Your example code needs one more check  to see if `uploaded_files['actual_files']` exists (isn't nil).